features/car/restrictions.feature

```diff
@@ -115,6 +115,36 @@ Feature: Car - Turn restrictions
             | c    | a  | cj,aj,aj |
             | c    | b  | cj,bj,bj |
 
+    @no_turning
+    Scenario: Car - No u-turn
+    # https://www.openstreetmap.org/edit?node=54878482#map=19/34.05242/-117.19067
+        Given the node map
+            """
+                c
+                3
+            a 1 x 2 b
+                4
+                d
+            """
+
+        And the ways
+            | nodes |
+            | ax    |
+            | xb    |
+            | cx    |
+            | xd    |
+
+        And the relations
+            | type        | way:from | way:to | node:via | restriction |
+            | restriction | ax       | ax     | x        | no_u_turn   |
+            | restriction | bx       | bx     | x        | no_u_turn   |
+            | restriction | cx       | cx     | x        | no_u_turn   |
+            | restriction | dx       | dx     | x        | no_u_turn   |
+
+        When I route I should get
```

Comment:
@mjjbell as you worked on PR closing https://github.com/Project-OSRM/osrm-backend/issues/4465 and this PR was blocked by it could you take a look? This test currently failing on master:
```
2022-10-06T14:33:17.0903046Z    Message:
2022-10-06T14:33:17.0903377Z      [31mTables were not identical:
2022-10-06T14:33:17.0903835Z      | waypoints |     route             |
2022-10-06T14:33:17.0904218Z      | a,x,a     | (-) ax,ax,xb,xb,ax,ax |
2022-10-06T14:33:17.0904606Z      | a,x,a     | (+) ax,xb,xb,xb,ax,ax |[39m
```

But it seems in this context `ax,ax,xb,xb,ax,ax` is the same as `ax,xb,xb,xb,ax,ax`, or not? Asking because tbh I don't have enough knowledge yet to now for sure if we can merge it or not(but probably would be great to merge it if it is ready) :)

Reply:
Short answer: yes

Longer answer:
Adding the `turns` to see what's going on.

```
| a,x,a     | (-) ax,ax,xb,xb,ax,ax | (-) |
| a,x,a     | (+) ax,xb,xb,xb,ax,ax | (+) depart,new name straight,continue uturn,arrive,depart,arrive |
```

So the response is splitting the route into two as  `a->x->b(uturn)->x , x->a`
whereas the test wants `a->x , x->b(uturn)->x->a`

They will have equivalent weight, so it's just whichever OSRM happens to select out of the two.

This wouldn't have worked in the past because the mid journey stop might have been on `cx` or somewhere else that wasn't optimal.